### DISCUSSION TITLE
Files in the fuzzy finder should be sorted by priority

### DISCUSSION THREAD

**Discussion by IgnisDa** [helix-editor/helix#10249] [Ideas]
Is there a way to have the fuzzy file finder remember priorities? I open some files more than other and I would like them to be at the top of the list. 

Then maybe we can expose an option in the configuration that enables this feature.

**Comment by sunnyzanchi** [comment]
i also think it'd be nice for the file finder to be a little smarter. i looked into the picker code and it looks like the [`fuzzy_match`](https://github.com/helix-editor/helix/blob/master/helix-term/src/ui/fuzzy_match.rs#L34) function could be changed to factor in a frecency score as well. i'd be happy to make a PR, i could try and implement something along [these lines](https://slack.engineering/a-faster-smarter-quick-switcher/). i'm not sure how Helix would handle persisting the history, direction in that regard would be appreciated 😄
